numpy/core/src/umath/fast_loop_macros.h

```diff
@@ -46,14 +46,26 @@ abs_ptrdiff(char *a, char *b)
     npy_intp i;\
     for(i = 0; i < n; i++, ip1 += is1, op1 += os1, op2 += os2)
 
-/** (ip1, ip2) -> (op1) */
-#define BINARY_LOOP\
+#define BINARY_LOOP_BASE\
     char *ip1 = args[0], *ip2 = args[1], *op1 = args[2];\
     npy_intp is1 = steps[0], is2 = steps[1], os1 = steps[2];\
     npy_intp n = dimensions[0];\
     npy_intp i;\
```

Comment:
This snippet reappears further down in the file - now that you've removed the `for` loop, you can reuse this macro in the definition of `BASE_BINARY_LOOP_INP`